test/test_unit.jl

```diff
@@ -0,0 +1,57 @@
+module TestUnit
+
+using Test
+using Trixi
+using TrixiShallowWater
+
+include("test_trixi.jl")
+
+# Start with a clean environment: remove Trixi.jl output directory if it exists
+outdir = "out"
+isdir(outdir) && rm(outdir, recursive = true)
+
+# Run various unit (= non-elixir-triggered) tests
+@testset "Unit tests" begin
+#! format: noindent
+
+@timed_testset "Shallow water conversion between conservative/entropy variables" begin
+    H, v, b = 3.5, 0.25, 0.4
+
+    let equations = ShallowWaterEquationsWetDry1D(gravity_constant = 9.8)
+        cons_vars = Trixi.prim2cons(SVector(H, v, b), equations)
+        entropy_vars = Trixi.cons2entropy(cons_vars, equations)
+        @test cons_vars ≈ Trixi.entropy2cons(entropy_vars, equations)
+
+        total_energy = Trixi.energy_total(cons_vars, equations)
+        @test total_energy ≈ Trixi.entropy(cons_vars, equations)
+        @test total_energy ≈
+              Trixi.energy_internal(cons_vars, equations) +
+              energy_kinetic(cons_vars, equations)
+        # test tuple args
+        cons_vars = Trixi.prim2cons((H, v, b), equations)
+        entropy_vars = Trixi.cons2entropy(cons_vars, equations)
+        @test cons_vars ≈ Trixi.entropy2cons(entropy_vars, equations)
+    end
+end
+
+# TODO: Probably can be removed. Since we dispatch to ShallowWaterEquations1D we don't need 
+#       equation specific wave speed estimates.
```

Comment:
But these wave speed estimate functions still have wrappers in the new package, right? So wouldn't we need this?

Reply:
This one is a bit confusing, but from what I can tell the wave speed estimates are only called from within the HLL and Lax-Friedrichs fluxes. As we introduce a wrapper function for these fluxes to the `ShallowWaterEquations1D` in Trixi.jl

`@inline function (numflux::Trixi.FluxHLL)(u_ll, u_rr, orientation_or_normal_direction,
                                          equations::ShallowWaterEquationsWetDry1D)
    return (numflux::Trixi.FluxHLL)(u_ll, u_rr, orientation_or_normal_direction,
                                    Trixi.ShallowWaterEquations1D(equations.gravity,
                                                                  equations.H0, eps(),
                                                                  eps()))
end` 

we never call the wave speed estimates directly from the new equations. So it is sufficient to have them only in Trixi.jl for the `ShallowWaterEquations1D`.

Reply:
I originally had them in the new package, but they are currently commented out and I am planning to remove them.

Reply:
The wave speed estimates are now removed, see 6b585b9296b27554032ddd30886d7f35349df6ed